### PR TITLE
homework db week2

### PR DESCRIPTION
**Pull request by panivita**

### CODE REVIEW

File: databases/week2/part4-database.sql [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+CREATE TABLE  `issue` (
+  `issue_id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `book_id` int(10) unsigned NOT NULL,
+  CONSTRAINT `fk_issue_book` FOREIGN KEY (`book_id`) REFERENCES `book` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+ `available_status` tinyint(1) NOT NULL DEFAULT '1',
+  `added_by` int(10) unsigned NOT NULL,
+   CONSTRAINT `fk_issue_staff` FOREIGN KEY (`added_by`) REFERENCES `staff` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  `added_at_timestamp` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```
Comment (flemeur) [inline databases/week2/part4-database.sql:57]: I'm not quite sure what the issue table is for. Is it for different copies of the same book or something else?

File: databases/week2/part4-database.sql [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+CREATE TABLE  `book_return` (
+  `return_id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `book_id` int(10) unsigned NOT NULL,
+  `student_id` int(10) unsigned NOT NULL,
+  CONSTRAINT `fk_return_book` FOREIGN KEY (`book_id`) REFERENCES `book` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_return_student` FOREIGN KEY (`student_id`) REFERENCES `student` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+ `issue_date` DATETIME NOT NULL,
+ `expiry_date` DATETIME NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```
Comment (flemeur) [inline databases/week2/part4-database.sql:67]: I might not understand this correctly, but it looks like this table holds all the times a book was borrowed by a student. If that is the case then i think the name `book_return` might be a bit misleading. Also how do you keep track of if and when a book has been returned?

Also if my assumption about the `issue` table is correct, then how do you keep track of which book copies have been borrowed?

File: databases/week2/part2-school.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+-- ALTER TABLE student
+-- ADD INDEX (name);
+
+-- 2.3 Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).
+ALTER TABLE class
+ADD Column status ENUM ('not-started', 'ongoing', 'finished');
+
+UPDATE class
+SET status = 'ongoing'
+WHERE id = 12 OR id = 13 OR id = 14;
```
[the remark this answers is not in the document]
Comment (panivita) [reply databases/week2/part2-school.sql:56]: Thanks. I changed that.
ALTER TABLE class
ALTER status SET DEFAULT 'ongoing';
Is it correct?

File: databases/week2/part2-school.sql [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+CREATE INDEX name_index
+ON student (name);
+
+-- OR
+-- ALTER TABLE student
+-- ADD INDEX (name);
+
+-- 2.3 Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).
+ALTER TABLE class
+ADD Column status ENUM ('not-started', 'ongoing', 'finished');
```
[the remark this answers is not in the document]
Comment (panivita) [reply databases/week2/part2-school.sql:52]: Thanks. I changed that to 
ALTER TABLE class
ADD Column status ENUM ('not-started', 'ongoing', 'finished') NOT NULL DEFAULT 'ongoing';
IS it correct?

[the remark this answers is not in the document]
Comment (flemeur) [reply databases/week2/part2-school.sql:52]: Yep, the value I was thinking of was `NULL`. And yes, looks correct with the default value as well. Whether the default value should be `not-started` or `ongoing` depends on the use case.

**Pull request merged**